### ISSUE TITLE
Clipboard error : Target STRING not available when running

### ISSUE THREAD

**Issue by edi9999** [neovim/neovim#2642]
Hi,

sometimes (I don't know exactly how to reproduce it) , when running the command

`:call getreg('*')`

I get the following error:

```
 clipboard: error: Error: target STRING not available
Error detected while processing function <SNR>13_YRCheckClipboard:
line   20:
clipboard: provider returned invalid data
Press ENTER or type command to continue
```

The call is made by the yankring plugin: https://github.com/vim-scripts/YankRing.vim/blob/master/plugin/yankring.vim#L910

The same command works well on vim, but not on nvim.

I've tried to use no vimrc with nvim -u NONE and I experience the same issue

I think this problem occurs when I copy something non texty in the clipboard

**Comment by justinmk** [comment]
> I think this problem occurs when I copy something non texty in the clipboard

That is quite likely. We need to handle that somehow.

**Comment by tracyone** [comment]
any process?This problem is still exist.

**Comment by cristicbz** [comment]
I can also repro this.

**Comment by nabn** [comment]
Any update on this? It's happens quite frequently for me.

**Comment by lusob** [comment]
I also can reproduce it, to fix it I copy something to the clipboard before starting neovim and the error disappears

**Comment by mhinz** [comment]
Do you all use yankring? I'd like to reproduce it with `nvim -u NONE`.

**Comment by edi9999** [comment]
You can with

:call getreg("*")

**Comment by mhinz** [comment]
@edi9999 Works for me in `nvim -u NONE`. `:reg *` shows the proper content.

**Comment by Pyrohh** [comment]
@edi9999 please post your nvim version too

**Comment by edi9999** [comment]
This seems to have been corrected a bit, it works quite fine with HTML copied from a page (in firefox).

The only problem (I don't know if it is expected to get an error), is when you copy only an image : 

   clipboard: error: Error: target STRING not available


```
 $ nvim --version
NVIM 0.1.0-dev (compiled Nov 18 2015 15:07:57)
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/build/config -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/src -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/.deps/usr/include -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/.deps/usr/include -I/usr/include/luajit-2.0 -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/.deps/usr/include -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/.deps/usr/include -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/.deps/usr/include -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/.deps/usr/include -I/usr/include -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/build/src/nvim/auto -I/build/neovim-CBgKsV/neovim-0.1.0ubuntu1+git201511181241+2002+15~ubuntu15.10.1/build/include
Compiled by buildd@lgw01-50

Optional features included (+) or not (-): +acl   +iconv    +jemalloc 
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

**Comment by lusob** [comment]
@edi9999 is working copying any text to system clipboard no just html, I tried it copying a text line from gedit and the problem disappears too. As you said if I copy an image the error appears again

**Comment by Pyrohh** [comment]
Huh, I can't seem to reproduce this using both `xclip` 0.12 and `xsel` 1.2.0p2 on OpenBSD. The version of X used there is a bit different than mainline Xorg, so maybe that affects things.

**Comment by emaniacs** [comment]
i get the same issue, to reproduce 
- take a screenshoot
- choose copy to clipboard
and then open neovim

**Comment by Pyrohh** [comment]
> take a screenshoot

using what program?

**Comment by emaniacs** [comment]
using default screenshoot apps in ubuntu

**Comment by azeey** [comment]
With only xsel installed (I removed xclip) I don't get this error anymore.

**Comment by stanislas-brossette** [comment]
I have the same issue. Using NVIM 0.1.4-dev
@mhinz The problem seems to be related to the use of the YankRing plugin.
I get the problem only when I activate YankRing.

**Comment by scrooloose** [comment]
This was also driving me crazy. So far, this has prevented the error:

```
"fix for yankring and neovim
let g:yankring_clipboard_monitor=0
```

**Comment by krisleech** [comment]
I get this error on every startup (even if I didn't yet copy anything to the system clipboard yet). The suggestion by @scrooloose fixes the issue for me and doesn't seem to have had any negative effects on my yankring usage thus far.

Ubuntu 16.02 using i3 Window Manager, not Unity.
NeoVim 0.1.6-dev
xclip 0.12

**Comment by justinmk** [comment]
@krisleech FWIW yankring is no longer needed, https://github.com/bfredl/nvim-miniyank uses the `TextYankPost` event to implement a true yankring (without mappings).

**Comment by erm3nda** [comment]
@azeey I got that problem just today running xclip from Python (it works with no problem at terminal). It worked for a while and suddently stopped to work. I could only find some threads with that strange bug. Didn't knew about `xsel` existance, no mention about alternatives when i search for. Thank you so much.

**Comment by erm3nda** [comment]
@azeey I got that problem just today running xclip from Python (it works with no problem at terminal). It worked for a while and suddently stopped to work. I could only find some threads with that strange bug. Didn't knew about `xsel` existance, no mention about alternatives when i search for. I tried it and worked for a while, then stopped to work again. So, finally i got to my clipboard manager tray icon, then flushed all, then started to work. Seems that is interfering in any way. Both `xclip` and `xsel` where interfered.

**Comment by justinmk** [comment]
This might be fixed since the clipboard now avoids repeated error messages. If not, open a new issue with the issue tempate filled out.

**Comment by snide** [comment]
On the off chance someone else still sees this error, you could be using an older yank plugin like https://github.com/svermeulen/vim-easyclip

Moving to something newer like https://github.com/svermeulen/vim-cutlass will avoid these issues.

**Comment by augustobmoura** [comment]
> On the off chance someone else still sees this error, you could be using an older yank plugin like [svermeulen/vim-easyclip](https://github.com/svermeulen/vim-easyclip)
> 
> Moving to something newer like [svermeulen/vim-cutlass](https://github.com/svermeulen/vim-cutlass) will avoid these issues.

I have the exact same problem when running https://github.com/svermeulen/vim-yoink, I don't think the issue is fixed on all modern plugins
